### DISCUSSION TITLE
How do I get GH Username based on Actions events

### DISCUSSION THREAD

**Discussion by byteshiva** [community/community#25196] [Actions] [answered]
<p>The below is the actions script where I need the username of github based on schedule event. </p>
<p>Under “Set git username and email”, How do I get the github username?</p>
<pre><code>name: Python application

on:
  schedule:
    # * is a special character in YAML so you have to quote this string
    - cron: '8 1 * * *'

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@master
    - name: Set git username and email
      run: |
        #
git config --global user.email "${GH_USERNAME}@users.noreply.github.com"
git config --global user.name "${GH_USERNAME}"
env:
GH_USERNAME: ${{ github.event.schedule.user.login }}
</code></pre>

**Comment by logankilpatrick** [comment]
<p>Hey <a class="mention" href="/byteshiva">@byteshiva</a>, I am a little confused. Is the existing setup you are using not working? If not, who’s username do you want? If the action is scheduled, then it’s not being triggered by a different user, so why not just hardcode it? </p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/byteshiva">@byteshiva</a> ,</p>
<p>You can check the github <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-printing-context-information-to-the-log-file" rel="nofollow noopener">context</a> to get the event info. For GH username, you can follow the context structure and get it with below code:</p>
<pre><code>${{ github.actor }}
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181089341-a7548634-53a9-4b84-b0c1-cce71245d482.png" alt="Scheduleactor2.png" data-base62-sha1="t3hADTcRS1MDr8VnRsgl08iKqIh" width="502" height="402"></p>
<p>There is no ‘user.login’ under ‘github.event.schedule’ from the context screenshot attached, that’s why you cannot get the value correctly.</p>
<p>Hope it helps!</p>

**Comment by byteshiva** [comment]
<p>Hey <a href="/t5/user/viewprofilepage/user-id/74324">@byteshiva</a>, I am a little confused. Is the existing setup you are using not working?</p>
<p>&gt;&gt; The setup I’ve posted is not working</p>
<p>If not, who’s username do you want?</p>
<p>&gt;&gt; I want to get owner/username of the repo.</p>
<p>If the action is scheduled, then it’s not being triggered by a different user, so why not just hardcode it? </p>
<p>&gt;&gt; I don’t want to hardcode, since it can forked and used by other users as well. The forked user should be able to get their information.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/byteshiva">@byteshiva</a> ,</p>
<p>The schedule event works on default branch(master), please make sure you check in correct branch.</p>
<p>Use ‘${{ github.actor }}’ instead of ‘${{ github.event.schedule.user.login }}’ in your code.</p>

**Comment by byteshiva** [comment, marked as answer]
<p>Thanks <a class="mention" href="/weide-zhou">@weide-zhou</a> , - I’ve added the ${{github.actor}} to get the username and have placed it in main action script and waiting for the job to be run. I’ve tested it in standalone scripts. its working.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/byteshiva">@byteshiva</a> ,</p>
<p>Thanks for your reply! Glad to know it works for you!</p>
<p>More info for you, if you’d like to schedule for other branches than default master, there’s a trick that add ‘<a href="https://github.com/actions/checkout#usage" rel="nofollow noopener">ref</a>’ parameter for the ‘actions/checkout’, specify the desired branch as the value. Hope it helps!</p>
<p>BTW, could you please help to mark then it can help others who has same query?</p>
<p>Thanks.</p>

**Reply by elibroftw** [reply]
@byteshiva please mark the other response as the answer not this one
